integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java

```diff
@@ -88,12 +89,23 @@ private static void populateCarbonTable(Configuration configuration, String path
         }
       }
     }
-    if (null != validInputPath) {
-      AbsoluteTableIdentifier absoluteTableIdentifier = AbsoluteTableIdentifier
-          .from(validInputPath, getDatabaseName(configuration), getTableName(configuration));
+    if (null != paths) {
       // read the schema file to get the absoluteTableIdentifier having the correct table id
       // persisted in the schema
-      CarbonTable carbonTable = SchemaReader.readCarbonTableFromStore(absoluteTableIdentifier);
+      CarbonTable carbonTable;
+      AbsoluteTableIdentifier absoluteTableIdentifier = AbsoluteTableIdentifier
+          .from(validInputPath, getDatabaseName(configuration), getTableName(configuration));
+      String schemaPath = CarbonTablePath.getSchemaFilePath(validInputPath);
+      if (FileFactory.getCarbonFile(schemaPath).exists()) {
+        // read the schema file to get the absoluteTableIdentifier having the correct table id
+        // persisted in the schema
+        carbonTable = SchemaReader.readCarbonTableFromStore(absoluteTableIdentifier);
+      } else {
+        // InferSchema from data file
+        carbonTable = CarbonTable.buildFromTableInfo(SchemaReader
+            .inferSchema(absoluteTableIdentifier, false));
+        carbonTable.setTransactionalTable(true);
```

Comment:
If shema doesn't exists, will it be non-transactional table ?

can you add comment why it is set as transactional ?

Reply:
it should be false.. changed it